piptools/resolver.py

```diff
@@ -87,14 +88,17 @@ def resolve(self, max_rounds=10):
 
     def _check_constraints(self):
         for constraint in chain(self.our_constraints, self.their_constraints):
-            if constraint.link is not None and not constraint.editable:
```

Comment:
This PR  is also about allowing editable requirements again, so yes, it was intentional.